go-controller/pkg/ovn/multi_homed_pods_test.go

```diff
@@ -0,0 +1,299 @@
+					return getPodAnnotations(fakeOvn.fakeClient.KubeClient, podInfo.namespace, podInfo.podName)
+				}).WithTimeout(2 * time.Second).Should(MatchJSON(podInfo.getAnnotationsJson()))
+
+				Eventually(fakeOvn.nbClient).Should(
+					libovsdbtest.HaveData(
+						append(
+							getExpectedDataPodsAndSwitches([]testPod{podInfo}, []string{nodeName}),
+							newSecondaryNetworkExpectationMachine(
+								fakeOvn,
+								[]testPod{podInfo},
+								expectationOptions...,
+							).expectedLogicalSwitchesAndPorts()...)))
+
+				return nil
+			}
+
+			Expect(app.Run([]string{app.Name})).To(Succeed())
+		},
+		table.Entry("pod with port security and not allowing unknown MAC addresses",
+			dummySecondaryNetInfo(),
+		),
+
+		table.Entry("pod without port security, and not allowing unknown MAC addresses",
+			dummySecondaryNetInfoWithoutPortSec(),
+			withoutPortSecurity(),
+		),
+
+		table.Entry("pod with port security, and allowing unknown addresses",
+			dummySecondaryNetInfoAllowingL2Traffic(),
+			allowSendingToUnknownL2Addrs(),
+		),
+
+		table.Entry("pod without port security and allowing unknown addresses",
+			dummySecondaryNetInfoWithoutPortSecAndAllowingL2Traffic(),
+			withoutPortSecurity(),
+			allowSendingToUnknownL2Addrs(),
+		),
+	)
+})
+
+func newMultiHomedTestPod(
+	nodeName, nodeSubnet, nodeMgtIP, nodeGWIP, podName, podIPs, podMAC, namespace string,
+	multiHomingConfigs ...secondaryNetInfo,
+) testPod {
+	pod := newTPod(nodeName, nodeSubnet, nodeMgtIP, nodeGWIP, podName, podIPs, podMAC, namespace)
+	for _, multiHomingConf := range multiHomingConfigs {
+		pod.addNetwork(
+			multiHomingConf.netName,
+			multiHomingConf.nadName,
+			"",
+			"",
+			"",
+			multiHomingConf.podIP,
+			multiHomingConf.podMAC,
+			0,
+		)
+	}
+	return pod
+}
+
+func newMultiHomedPod(namespace, name, node, podIP string, multiHomingConfigs ...secondaryNetInfo) *v1.Pod {
+	pod := newPod(namespace, name, node, podIP)
+	var secondaryNetworks []nadapi.NetworkSelectionElement
+	for _, multiHomingConf := range multiHomingConfigs {
+		nadNamePair := strings.Split(multiHomingConf.nadName, "/")
+		ns := pod.Namespace
+		attachmentName := multiHomingConf.nadName
+		if len(nadNamePair) > 1 {
+			ns = nadNamePair[0]
+			attachmentName = nadNamePair[1]
+		}
+		nse := nadapi.NetworkSelectionElement{
+			Name:       attachmentName,
+			Namespace:  ns,
+			MacRequest: multiHomingConf.podMAC,
+		}
+		if len(multiHomingConf.podIP) > 0 {
+			nse.IPRequest = []string{multiHomingConf.podIP + "/24"}
+		}
+		secondaryNetworks = append(secondaryNetworks, nse)
+	}
+	serializedNetworkSelectionElements, _ := json.Marshal(secondaryNetworks)
+	pod.Annotations = map[string]string{nadapi.NetworkAttachmentAnnot: string(serializedNetworkSelectionElements)}
+	return pod
+}
+
+func namespacedName(ns, name string) string { return fmt.Sprintf("%s/%s", ns, name) }
+
+func (sni *secondaryNetInfo) setupOVNDependencies(dbData *libovsdbtest.TestSetup, netConf ovncnitypes.NetConf) error {
```

Comment:
isn't the `netConf` parameter here derived from the `secondaryNetInfo` itself? So potentially we would not need to pass it?

Reply:
Done.